src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java

```diff
@@ -67,4 +69,9 @@ public boolean isFormattedConfig() {
   public boolean hasFormattedInitValue() {
     return entityName() != null && value() != null;
   }
+
+  private static String parseValueString(String valueString) {
+    String randomValue = Integer.toString(Math.abs(valueString.hashCode()));
```

Comment:
This approach will produce the same "random" value for the same string. Is that what we want? The most common case will be just the string "$RANDOM", which is likely to occur often and will always produce the same value.

Is this a problem? Or is it more important for the value to be reproducible?